src/main/kotlin/io/zerobase/smarttracing/resources/OrganizationsResource.kt

```diff
@@ -64,38 +66,46 @@ class OrganizationsResource(val dao: GraphDao, private val siteTypes: MultiMap<S
         val hasTestingFacilities = request.hasTestingFacilities ?: false
         val hasMultipleSites = request.hasMultipleSites ?: true
 
-        val id = dao.createOrganization(
-            name, phone, email,
-            contactName, address,
-            hasTestingFacilities,
-            hasMultipleSites
-        )
-
-        return when (id) {
-            null -> null
-            else -> IdWrapper(id)
+        try {
+            val id = dao.createOrganization(
+                name, phone, email,
+                contactName, address,
+                hasTestingFacilities,
+                hasMultipleSites
+            )
+            return id?.let { IdWrapper(id) }
+        } catch (e: Exception) {
+            val res = Response.status(Response.Status.BAD_REQUEST)
+                .entity(e.message)
+                .build()
+            throw WebApplicationException(res)
         }
     }
 
     @Path("/{id}/sites")
     @POST
     @Creator
+    @SuppressFBWarnings("RCN_REDUNDANT_NULLCHECK_OF_NONNULL_VALUE", justification = "false positive")
     fun createSite(@PathParam("id") id: String, request: CreateSiteRequest): IdWrapper? {
         val name = request.name ?: ""
         val category = request.category
         val subcategory = request.subcategory
         val latitude = request.location.latitude
         val longitude = request.location.longitude
         val isTesting = request.isTesting
-        val isCat = siteTypes.containsKey(category)
-        val isSubCat = siteTypes[category]?.contains(subcategory)
 
-        if (!(isCat!!)) {
-            return null
+        if (!(siteTypes.containsKey(category))) {
+            val res = Response.status(Response.Status.BAD_REQUEST)
+                .entity("Not a valid category please check /models/site-types")
+                .build()
+            throw WebApplicationException(res)
         }
 
-        if (!(isSubCat!!)) {
-            return null
+        if (!(siteTypes[category]?.contains(subcategory) ?: false)) {
```

Comment:
nit: unnecessary parenthesis

Reply:
would that not change the false to a true?

Reply:
you don't need the safe qualifier on this one because you already validated the key exists. you can either use the `!!` (this is a place where it's acceptable) or you can make a temp variable for the subcategories 

```kotlin
val subcategories = siteTypes[category]
if (subcategories == null ) {
  // return/throw error
}
if (!subcategories.contains(subcategories)) {
  // return/throw error
}
```

Reply:
oh nevermind i see why that part does not really matter